### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by rieske**
- add some explanation about what another CA plugin solves
- fix the publishing application code snippet
- add a plugin portal badge
- don't trigger a release on README change

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,25 @@
 # codeartifact-gradle-plugin
 
-A set of plugins for authenticating with and using [AWS CodeArtifact](https://aws.amazon.com/codeartifact/) repository as a source for project plugins 
+[![Gradle Plugin Portal](https://img.shields.io/maven-metadata/v/https/plugins.gradle.org/m2/co/bound/plugins/maven-metadata.xml.svg?colorB=007ec6&label=Plugin%20Portal)](https://plugins.gradle.org/plugin/co.bound.codeartifact)
+
+A set of plugins for authenticating with and using [AWS CodeArtifact](https://aws.amazon.com/codeartifact/) repository as a source for project plugins
 and dependencies as well as target for publishing artifacts to.
```
Comment (rnunn-bound) [inline README.md:6]: "as well as _a_ target"?

File: README.md [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
 
+## Why another CodeArtifact plugin?
+
+The existing solutions for Gradle with CodeArtifact cater for the most common use cases - consuming and publishing
+project dependencies from/to CodeArtifact repository.
+
+Another use case exists though - organizations that work with multiple projects scattered in multiple repositories may want
```
Comment (rnunn-bound) [inline README.md:13]: "work with multiple projects spread across many repositories"? "Scattered" sounds messy and chaotic, which is probably not the impression we want to give 😄

File: README.md [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+to [encapsulate aspects of common project conventions and build logic using Gradle plugins](https://docs.gradle.org/current/samples/sample_publishing_convention_plugins.html).
+Those plugins are published to a repository like any other libraries, however, to consume them,
```
Comment (rnunn-bound) [inline README.md:15]: Either "like other libraries" or "like any other library". I'd probably start a new sentence with "However, to consume them, Gradle needs...".

**Comment by rieske** [comment]
Thanks @rnunn-bound

**Review by rnunn-bound — APPROVED** [review]

**Pull request merged**